### PR TITLE
feat: implemented separate thread for printing status

### PR DESCRIPTION
**Pull request by AntoniaSzecsi**

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -1,11 +1,42 @@
...
+        ready = true;
+    }
+
+    std::thread printingThread(statusThread);
+    printingThread.detach();
+
+    // I noticed in multiple examples that an instruction like this keeps the main thread alive and prevents it from exiting, serving as a placeholder
+    while (true) {
+        std::this_thread::sleep_for(std::chrono::seconds(1)); 
+    }
```
Comment (CristianIonutIonescu) [inline main.cc]: try using 
````
printringThread.join();
````
to block the main thread.

File: main.cc [hunk 2]
```diff
@@ -1,11 +1,42 @@
...
+int main() {
+    smart_home::User::getInstance().controlDevicesDemo();
+    smart_home::User::getUserHouse().writeDataToFile();
+
+    {
+        std::lock_guard<std::mutex> lock(readyMutex);
+        ready = true;
+    }
+
+    std::thread printingThread(statusThread);
```
Comment (CristianIonutIonescu) [inline main.cc]: you display on 2 different threads but the action of the main thread is finished by the time you create the display thread.
ControlDevicesDemo should be called during the printing thread run.

**Review by CristianIonutIonescu — CHANGES_REQUESTED** [review]
Please address the 2nd comment especially

File: main.cc (hunk 2 above)
Comment (AntoniaSzecsi) [reply main.cc]: ok so basically they have to run simultaneously, on parallel threads. makes more sense like this:)

**Review by CristianIonutIonescu — APPROVED** [review]

**Pull request merged**